Tiltfile

```diff
@@ -395,16 +532,37 @@ def prepare_all():
                 context = context,
             )
 
+    addons_arg = ""
+    for name in settings.get("deploy_addons", []):
+        addon = addons.get(name)
+        if addon == None:
+            fail("Addon with name " + name + " not found")
+        if addon.get("kustomize_config", True):
+            addon_substitutions = addon.get("kustomize_substitutions", {})
+            for addon_substitution in addon_substitutions:
+                if addon_substitution in kustomize_substitutions:
```

Comment:
I think we should fail only if two addons are setting the same variable to different values.
FYI we can probably get rid of this limitation when we will merge https://github.com/kubernetes-sigs/cluster-api/pull/6483, where we can control to a more granular level which env vars are used by each customize call.